### ISSUE TITLE
Add a option to allow the engine update vulnerabilities database through proxy

### ISSUE THREAD

**Issue by JayceCao** [anchore/anchore-engine#300]
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one): FEATURE REQUEST

**Version of Anchore Engine and Anchore CLI if applicable**:

 
**What happened**:

**What did you expect to happen**:

**Any relevant log output from /var/log/anchore**:

**What docker images are you using**:

**How to reproduce the issue**:

**Anything else we need to know**:

**Comment by CodingJzy** [comment]
This is right, our company is like this, can't connect. However, proxy resolution can also be set inside the container.

**Comment by JayceCao** [comment]
It seems that if configuring `HTTP_PROXY`, `HTTPS_PROXY` or `ALL_PROXY` for containers, all http connections would use the proxy, including internal connections. It's annoyed that `NO_PROXY` doesn't support CIDR (network address like 10.244.0.0/16)
> This is right, our company is like this, can't connect. However, proxy resolution can also be set inside the container.

**Comment by CodingJzy** [comment]
是。而且。我们的项目无法联网。就算设置代理也没用。

**Comment by zhill** [comment]
Understood. Thanks!

**Comment by judexzhu** [comment]
I also wish this feature can be applied soon, thanks

**Comment by SysEngDan** [comment]
The standard environment variables work just fine. I am behind a proxy and I have configured the entire stack to access the internet through the proxy. I added this to my docker-compose.yml file:

```
    environment:
      - HTTP_PROXY=http://myproxy.com:80
      - HTTPS_PROXY=http://myproxy.com:80
      - http_proxy=http://myproxy.com:80
      - https_proxy=http://myproxy.com:80
      - NO_PROXY=queue:8228,api:8228,catalog:8228,policy-engine:8228,analyzer:8228,*.mycompany.com,10.0.0.0/8,127.0.0.1,172.0.0.0/8,localhost
      - no_proxy=queue:8228,api:8228,catalog:8228,policy-engine:8228,analyzer:8228,*.mycompany.com,10.0.0.0/8,127.0.0.1,172.0.0.0/8,localhost
```
